event-svc/src/event/service.rs

```diff
@@ -366,7 +385,14 @@ impl EventService {
                             event_cid,
                             init,
                             previous: vec![*data.prev()],
-                            data: data.data().to_json_bytes().map_err(|e| {
+                            data: DataContainer::new_with_should_index(
+                                data.header()
+                                    .map(|header| header.should_index())
+                                    .unwrap_or(SHOULD_INDEX_DEFAULT),
```

Comment:
I think if the field is absent in the data event we could just omit it from the corresponding conclusion event.  WDYT?

Reply:
Depends on the intended behavior. What does it mean for it to be absent? I don't like the idea of rolling up from the init event each time because that implies that any metadata changes revert to init if they are not specified. Where I would expect the behavior to be that it doesn't change from the previous event if its absent. Seems we need to design this a bit to get a clear answer.